openpype/hosts/blender/api/plugin.py

```diff
@@ -144,20 +147,196 @@ def deselect_all():
 
-        if (self.options or {}).get("useSelection"):
-            for obj in get_selection():
-                collection.objects.link(obj)
+        # Cache subsets in shared data
+        self.cache_subsets(self.collection_shared_data)
+
+        # Get cached subsets
+        cached_subsets = self.collection_shared_data.get(
+            "blender_cached_subsets"
+        )
+        if not cached_subsets:
+            return
+
+        # Process only instances that were created by this creator
+        for instance_node in cached_subsets.get(self.identifier, []):
+            property = instance_node.get(AVALON_PROPERTY)
+            # Create instance object from existing data
+            instance = CreatedInstance.from_existing(
+                instance_data=property.to_dict(),
+                creator=self
+            )
+            instance.transient_data["instance_node"] = instance_node
+
+            # Add instance to create context
+            self._add_instance_to_context(instance)
+
+    def update_instances(self, update_list):
+        """Override abstract method from BaseCreator.
+        Store changes of existing instances so they can be recollected.
+
+        Args:
+            update_list(List[UpdateData]): Changed instances
+                and their changes, as a list of tuples."""
+        for created_instance, _changes in update_list:
+            data = created_instance.data_to_store()
+            node = created_instance.transient_data["instance_node"]
+            if node:
+                imprint(node, data)
+
+    def remove_instances(self, instances: List[CreatedInstance]):
+
+        for instance in instances:
+            node = instance.transient_data["instance_node"]
+
+            if isinstance(node, bpy.types.Collection):
+                for children in node.children_recursive:
+                    if isinstance(children, bpy.types.Collection):
+                        bpy.data.collections.remove(children)
+                    else:
+                        bpy.data.objects.remove(children)
+
+                bpy.data.collections.remove(node)
+            elif isinstance(node, bpy.types.Object):
+                bpy.data.objects.remove(node)
+
+            self._remove_instance_from_context(instance)
+
+    def set_instance_data(
+        self,
+        subset_name: str,
+        instance_data: dict
+    ):
+        """Fill instance data with required items.
+
+        Args:
+            subset_name(str): Subset name of created instance.
+            instance_data(dict): Instance base data.
+            instance_node(bpy.types.ID): Instance node in blender scene.
+        """
+        if not instance_data:
+            instance_data = {}
+
+        instance_data.update(
+            {
+                "id": "pyblish.avalon.instance",
+                "creator_identifier": self.identifier,
+                "label": subset_name,
+                "task": get_current_task_name(),
```

Comment:
Why is this here?

Reply:
Well spotted - should be removed I suspect. Investigating.

Reply:
Removed with https://github.com/ynput/OpenPype/pull/5910/commits/0556986a3e39c45be8c923f18a87981c17163c3a